source/DSCResources/DSC_CMSiteMaintenance/DSC_CMSiteMaintenance.psm1

```diff
@@ -0,0 +1,578 @@
+        Not used in Get-TargetResource.
+
+    .NOTES
+        Device Type specifies what params are avalible TaskType 1 is backup TaskType 2 is SummaryTask and 3 is MaintenanceTask
+        TaskType 1 requires DeviceName (backup location) TaskType 2 may set run now and RunIntervalMins and TaskType 3 is standard
+        Maintenance Tasks.  On Type 1 and 2 DeleteOlderThan have a value of 0.  TaskName of Clear Install Flag in the console equals
+        'Clear Undiscovered Clients'.
+#>
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteCode,
+
+        [Parameter(Mandatory = $true)]
+        [ValidateSet('Delete Aged Inventory History','Delete Aged Metering Data','Clear Undiscovered Clients','Delete Obsolete Alerts',
+        'Delete Aged Replication Data','Delete Aged Device Wipe Record','Delete Aged Enrolled Devices','Delete Aged User Device Affinity Data',
+        'Delete Duplicate System Discovery Data','Delete Aged Unknown Computers','Delete Expired MDM Bulk Enroll Package Records','Backup SMS Site Server',
+        'Delete Aged Status Messages','Delete Aged Metering Summary Data','Delete Inactive Client Discovery Data',
+        'Delete Aged Application Revisions','Delete Aged Replication Summary Data','Delete Obsolete Forest Discovery Sites And Subnets',
+        'Delete Aged Threat Data','Delete Aged Delete Detection Data','Delete Aged Distribution Point Usage Stats',
+        'Delete Orphaned Client Deployment State Records','Rebuild Indexes','Delete Aged Discovery Data','Summarize File Usage Metering Data',
+        'Delete Obsolete Client Discovery Data','Delete Aged Log Data','Delete Aged Application Request Data',
+        'Check Application Title with Inventory Information','Delete Aged EP Health Status History Data','Delete Aged Notification Task History',
+        'Delete Aged Passcode Records','Delete Aged Console Connection Data','Monitor Keys','Delete Aged Collected Files',
+        'Summarize Monthly Usage Metering Data','Delete Aged Computer Association Data','Delete Aged Client Download History',
+        'Delete Aged Exchange Partnership','Summarize Installed Software Data','Delete Aged Client Operations',
+        'Delete Aged Notification Server History','Update Application Available Targeting',
+        'Delete Aged Cloud Management Gateway Traffic Data','Update Application Catalog Tables')]
+        [String]
+        $TaskName,
+
+        [Parameter(Mandatory = $true)]
+        [Boolean]
+        $Enabled
+    )
+
+    Write-Verbose -Message $script:localizedData.RetrieveSettingValue
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+
+    if ($TaskName -ne 'Update Application Catalog Tables')
+    {
+        $siteMaintenance = (Get-CMSiteMaintenanceTask -Name $TaskName -SiteCode $SiteCode).ManagedObject
+
```

Comment:
Add if ($siteMaintenance) here so get won't throw if the task type doesn't exist for the site (CAS vs PRI)